datastore.go

```diff
@@ -29,13 +29,26 @@ type Datastore struct {
 var _ ds.Datastore = (*Datastore)(nil)
 var _ ds.Batching = (*Datastore)(nil)
 
+var defaultSplit = func(a []byte) int {
+	return len(a)
+}
+
 // NewDatastore creates a pebble-backed datastore.
 func NewDatastore(path string, opts *pebble.Options) (*Datastore, error) {
 	if opts == nil {
 		opts = &pebble.Options{}
 		opts.EnsureDefaults()
 	}
 	opts.Logger = logger
+	// We force a default Split function that enables using bloom filters
+	// on lookups. Normally, our datastore keys are not versioned and
+	// correspond to unique items (cids) rather than MVCC keys.  On the
+	// other side, we expect a decent number of Has() and Get() calls with
+	// negative results, and those are currently very expensive and
+	// trigger a fair amount of reads. See
+	// https://github.com/cockroachdb/pebble/issues/2369#issuecomment-1450997680
+	opts.Comparer.Split = defaultSplit
```

Comment:
There are quite a few options in pebble and I wonder if forcing fallback on this is not always ideal.

Would it be reasonable to:
*  fall back on this _if `opts` is unset_, and 
* move the inline comments above to `godoc` in order to offer guidance in a case where options are non-nil

Reply:
I think I'm going to wait for someone to actually come and say they need something else and that a custom Split should be allowed.

The commend should def be moved!